streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -269,6 +270,39 @@ class StreamingContext private[streaming] (
     RDDOperationScope.withScope(sc, name, allowNesting = false, ignoreParent = false)(body)
   }
 
+  private[streaming] val recoverableAccuNameToId: mutable.Map[String, Long] = {
+    if (isCheckpointPresent) {
+      // accumulators created by StreamingContext must provide name, so it's safe to call name.get
+      mutable.Map(_cp.trackedAccs.map(accCP => (accCP.name, -1L)).toSeq: _*)
+    } else {
+      mutable.Map.empty
+    }
+  }
+
+  /**
+    * Different from accumulator in SparkContext, it will first try to recover from Checkpoint
+    * if it exist.
+    *
+    * @param initialValue   initial value of accumulator. It will be ignored when recovering from
+    *                       checkpoint
+    * @param name           name is required as identity to find corresponding accumulator.
+    */
+  def accumulator[T](initialValue: T, name: String)(implicit param: AccumulatorParam[T])
```

Comment:
hmm... change func name might me a good idea to explicitly emphasize the recoverable feature. I will change it. 

About the input parameter, I don't think `createFunc `here is necessary, `initialValue` should be enough.